lib/determinator/feature.rb

```diff
@@ -72,14 +73,43 @@ def parse_target_group(target_group)
 
       TargetGroup.new(
         rollout: target_group['rollout'].to_i,
-        constraints: constraints.each_with_object({}) do |(key, value), hash|
-          hash[key.to_s] = [*value].map(&:to_s)
-        end
+        constraints: parse_constraints(constraints)
       )
 
     # Invalid target groups are ignored
     rescue
       nil
     end
+
+    def parse_fixed_determinations(fixed_determinations)
+      fixed_determinations.map(&method(:parse_fixed_determination)).compact
+    end
+
+    def parse_fixed_determination(fixed_determination)
+      return fixed_determination if fixed_determination.is_a? FixedDetermination
+
+      variant = fixed_determination['variant']
+      raise if variant && !variants.keys.include?(variant)
```

Comment:
I'm not sure if these validations are really needed... It might make more sense to be liberal with what we accept here, and enforce the constraints in the UI, considering that we still have a deterministic way of handling the features if these conditions are not satisfied.

If we do want to keep them, I'd just return `nil` - no need to use exceptions for control flow.